shadows/framework/src/main/java/org/robolectric/shadows/ShadowRegion.java

```diff
@@ -1,17 +1,48 @@
 package org.robolectric.shadows;
 
+import static org.robolectric.util.reflector.Reflector.reflector;
+
 import android.graphics.Region;
 import org.robolectric.RuntimeEnvironment;
 import org.robolectric.annotation.HiddenApi;
 import org.robolectric.annotation.Implementation;
 import org.robolectric.annotation.Implements;
+import org.robolectric.annotation.RealObject;
+import org.robolectric.util.reflector.Direct;
+import org.robolectric.util.reflector.ForType;
 
 @Implements(Region.class)
 public class ShadowRegion {
+  @RealObject Region realRegion;
+
   public static long nextId = 1;
 
-  @HiddenApi @Implementation
-  public static Number nativeConstructor() {
+  /**
+   * The real {@link Region#equals(Object)} calls into native code, which is a no-op in Robolectric,
+   * and will always return false no matter what is compared. We can special-case some simple
+   * scenarios here.
+   */
+  @Implementation
+  @SuppressWarnings("EqualsHashCode")
+  public boolean equals(Object obj) {
+    if (!(obj instanceof Region)) {
+      return false;
+    }
+    if (obj == realRegion) {
```

Comment:
What about moving reference checking to the first line?

Reply:
What's the advantage of doing that (just curious) ?

Reply:
We can return true quicker when comparing itself. Most of time, we call `Region#equals` with a `Region` instance. But I don't think it can improve visual performance, even though slight performance improvement. Let's omit it.